src/program.js

```diff
@@ -67,6 +69,13 @@ export class Program {
     name: string, description: string, executor: Function,
     commandOptions: Object = {}
   ): Program {
+    this.mapCommandToSubOpts = {};
+    this.mapCommandToSubOpts[name] = commandOptions;
+    this.commandExecuted = name;
+    const subCommandDefaultValues = setDefaultValues(commandOptions);
+    this.defaultValues = {...this.defaultValues,
+      ...subCommandDefaultValues};
```

Comment:
While technically unlikely, a sub-command option default named `sourceDir` would overwrite the global value named `sourceDir`. This would be confusing and I don't think it's necessary. You should be able to do this, no?

````js
this.defaultValues[name] = setDefaultValues(commandOptions);
````